Captcha support

yunzhu-li opened the pull request:
Closes #21 

This is pretty ugly, but works for now.

![screenshot_1502514489](https://user-images.githubusercontent.com/1742004/29238082-17dee87c-7efb-11e7-9e39-fc46817c8396.png)

![screenshot_1502514501](https://user-images.githubusercontent.com/1742004/29238083-19d1156a-7efb-11e7-9efe-0bae9959584d.png)

meiyan-xie commented:
1. Can login even didn't input Captcha or input a wrong Captcha.
<img width="321" alt="screen shot 2017-08-12 at 19 37 05" src="https://user-images.githubusercontent.com/20272529/29240450-31f6af04-7f98-11e7-91ac-9072a32756b7.png">

2. After login, didn't show the user name. The login button still is "点此登录"
<img width="326" alt="screen shot 2017-08-12 at 19 40 11" src="https://user-images.githubusercontent.com/20272529/29240424-955a2306-7f97-11e7-8007-d953d0b06dd5.png">

3. It takes too long for loading the Captcha when click "点此登录".

4. 当光标移到填写验证码空格时，无法通过按Enter键将光标移到“登录”按钮。

5. Overall, you did really good! Thumb up!

yunzhu-li commented:
Fixed a couple of issues:

> Can login even didn't input Captcha or input a wrong Captcha

If you use our demo account, captcha value is ignored, otherwise, captcha validation is done by the school system. You can test with one of the real user accounts listed here: https://github.com/njitdev/sa-api

> After login, didn't show the user name. The login button still is "点此登录"

This is a bug when demo_account is used, now fixed and added a progress bar to indicate status.

> It takes too long for loading the Captcha when click "点此登录".

It normally takes around 2 seconds to complete the initialization and captcha fetching. It requires 2 sequential requests to the school system and it'll be slower when testing with `dev` backend from China since `dev` backend is in US.

> 当光标移到填写验证码空格时，无法通过按Enter键将光标移到“登录”按钮。

Fixed, when you hit "Enter" at captcha text box, it will submit directly.
